metpy/gridding/gridding_functions.py

```diff
@@ -182,7 +186,10 @@ def interpolate(x, y, z, interp_type='linear', hres=50000,
         2-dimensional array representing the interpolated values for each grid.
 
     """
-    grid_x, grid_y = points.generate_grid(hres, points.get_boundary_coords(x, y))
+    if boundary_coords is None:
```

Comment:
How about instead:
```python
if boundary_coords is None:
    boundary_coords = points.get_boundary_coords(x, y)
grid_x, grid_y = points.generate_grid(hres, boundary_coords)
```
?

Reply:
Less fluff, I like it! Commit coming.

Reply:
The important part to me was to reduce duplication, so that there's only one call to `generate_grid` that needs maintaining.